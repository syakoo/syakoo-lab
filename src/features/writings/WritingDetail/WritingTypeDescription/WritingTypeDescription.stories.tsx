import type { Meta, StoryObj } from "@storybook/react";

import { WritingTypeDescription } from ".";

const meta = {
  component: WritingTypeDescription,
  parameters: {
    layout: "fullscreen",
  },
} satisfies Meta<typeof WritingTypeDescription>;

export default meta;

type Story = StoryObj<typeof meta>;
export const Sample: Story = {
  args: {
    type: "article",
  },
};
