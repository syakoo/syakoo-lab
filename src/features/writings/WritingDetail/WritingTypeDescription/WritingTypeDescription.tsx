import { Icon } from "@/design-system/icons";
import { Col, Row } from "@/design-system/layout";
import { tokens } from "@/design-system/tokens";
import { P, Text } from "@/design-system/ui";
import type { WritingType } from "@/features/writings/_models/types";
import { writingTypeConfig } from "@/features/writings/_models/writingType";

type WritingTypeDescriptionProps = {
  type: WritingType;
};

export const WritingTypeDescription: React.FC<WritingTypeDescriptionProps> = ({
  type,
}) => {
  const { iconName, label, description } = writingTypeConfig.find(
    (config) => config.type === type,
  )!;

  return (
    <Col as="section" gap="50">
      <Row align="center" gap="50">
        <Icon name={iconName} width={tokens.fontSizes[200]} />
        <Text weight="bold">{label}</Text>
      </Row>
      <div>
        <P color="secondary" size="75">
          {description}
        </P>
      </div>
    </Col>
  );
};
