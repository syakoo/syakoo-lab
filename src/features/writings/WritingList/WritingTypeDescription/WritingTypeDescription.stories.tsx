import type { Meta, StoryObj } from "@storybook/react";
import { WritingTypeDescription } from ".";

const meta = {
  component: WritingTypeDescription,
} satisfies Meta<typeof WritingTypeDescription>;

export default meta;

type Story = StoryObj<typeof meta>;
export const Sample: Story = {
  args: {
    type: "all",
  },
};

export const All = {
  render: () => {
    return (
      <div>
        <WritingTypeDescription type="all" />
        <WritingTypeDescription type="article" />
        <WritingTypeDescription type="note" />
        <WritingTypeDescription type="diary" />
      </div>
    );
  },
};
