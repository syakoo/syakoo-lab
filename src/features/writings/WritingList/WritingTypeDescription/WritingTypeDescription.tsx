import type { WritingListType } from "../_shared/writingListType";
import { Text } from "@/design-system/ui";

type WritingTypeDescriptionProps = {
  type: WritingListType;
};

export const WritingTypeDescription: React.FC<WritingTypeDescriptionProps> = ({
  type,
}) => {
  switch (type) {
    case "all":
      return null;
    case "article":
      return <Text>Article は気合いの入った書き物を残しています。</Text>;
    case "note":
      return (
        <Text>
          Note はその時の思いつきやメモなど、アウトプットを目的としています。
        </Text>
      );
    case "diary":
      return (
        <Text>
          Diary は著者の身の回りで起こったことや全く中身のない話を残しています。
        </Text>
      );
  }
};
