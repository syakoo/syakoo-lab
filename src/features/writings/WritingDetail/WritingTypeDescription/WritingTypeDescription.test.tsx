import { testStories } from "@/test/testStories";

import * as stories from "./WritingTypeDescription.stories";

describe("WritingTypeDescription", () => {
  testStories(stories);
});
