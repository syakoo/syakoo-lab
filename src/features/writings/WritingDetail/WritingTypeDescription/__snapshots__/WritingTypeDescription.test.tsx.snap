// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WritingTypeDescription スナップショット Sample 1`] = `
<body>
  <div>
    <section
      class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_column__15ivi84k Flex_flexStyle_gap_50__15ivi84x Flex_flexStyle_variants_gap_50__15ivi84c Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
    >
      <div
        class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_center__15ivi84s Flex_flexStyle_variants_align_center__15ivi847 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_50__15ivi84x Flex_flexStyle_variants_gap_50__15ivi84c Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 14.25v-2.625a3.375 3.375 0 00-3.375-3.375h-1.5A1.125 1.125 0 0113.5 7.125v-1.5a3.375 3.375 0 00-3.375-3.375H8.25m0 12.75h7.5m-7.5 3H12M10.5 2.25H5.625c-.621 0-1.125.504-1.125 1.125v17.25c0 .621.504 1.125 1.125 1.125h12.75c.621 0 1.125-.504 1.125-1.125V11.25a9 9 0 00-9-9z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <p
          class="Text_textStyle__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_100__jzpfqdl Text_textStyle_variants_size_100__jzpfqd6 Text_textStyle_weight_bold__jzpfqdt"
        >
          Article
        </p>
      </div>
      <div>
        <p
          class="Text_textStyle__jzpfqde Text_textStyle_color_secondary__jzpfqdg Text_textStyle_variants_color_secondary__jzpfqd1 Text_textStyle_size_75__jzpfqdk Text_textStyle_variants_size_75__jzpfqd5 Text_textStyle_weight_normal__jzpfqdu"
        >
          Article は気合いの入った文書を残しています。
        </p>
      </div>
    </section>
  </div>
</body>
`;
